### PR TITLE
Test for #1979

### PR DESCRIPTION
**Pull request by jogibear9988**

### CODE REVIEW

**Comment by sdanyliv** [comment]
What are you trying to achieve here?
Getting records from table `issues` and use association for table `taggings`? I can not get whole idea of this example.

**Comment by jogibear9988** [comment]
I expect it to create a query like this...

	select users.lastname, tags.name from issues
	left join users on issues.assigned_to_id = users.id
	left join (select * from taggings where taggable_type = 'Issue' ) as ab on ab.taggable_id = issues.id
	left join tags on tags.id = ab.tag_id
	where tags.name = 'Visu'

**Comment by jogibear9988** [comment]
@sdanyliv is this possible? or what did i do wrong?

**Comment by sdanyliv** [comment]
@jogibear9988, i don't think so. 
Try to write real query using linq, without associations maybe it may help to understand how to do that properly.

**Comment by jogibear9988** [comment]
I did this...

But I think it's not the same...

        var query = from i in da.Issues
                    join t in (da.Taggings.Where(x => x.TaggableType == "Issue")) on i.Id equals t.TaggableId into tj
                    from t in tj.DefaultIfEmpty()
                    join tg in da.Tags on t.TagId equals tg.Id into tgj
                    from tg in tgj.DefaultIfEmpty()
                    join u in da.Users on i.AssignedToId equals u.Id into uj
                    from u in uj.DefaultIfEmpty()
                    where tg.Name == "Visu"
                    select new { Issue = i, User = u };

**Comment by jogibear9988** [comment]
Cause I only want each issue once, and so it could be returned more often.

**Comment by sdanyliv** [comment]
Something like this?
```cs
 var query = 
    from i in da.Issues
    from u in da.Users.LeftJoin(u => u.AssignedToId == u.Id)
    from ab in (da.Taggings.Where(x => x.TaggableType == "Issue")).LeftJoin(ab => ab.TaggableId == i.Id)
    from tg in da.Tags.LeftJoin(tg =>tg.Id == ab.TagId) 
    where tg.Name == "Visu"
    select new { Issue = i, User = u };
```

One note, if you have filter `tg.Name == "Visu"`, you do not need LeftJoin

**Comment by jogibear9988** [comment]
And is this possible with my associations?

**Comment by jogibear9988** [comment]
@sdanyliv I've added the query I expect...

**Comment by MaceWindu** [comment]
- reduced test code to contain only required details
- added linq-based test to demonstrate expectations
- fixed couple of isses with association test, but it still fails, I suspect something missing in handling of nested association contexts

@jogibear9988 in future, try to simplify tests to show linq2db issue, not your application issue :smile:

File: Source/LinqToDB/Linq/Builder/TableBuilder.TableContext.cs [hunk 1]
```diff
@@ -1420,7 +1395,7 @@ select om
 					isNew = true;
 				}
 
-				if (tableAssociation == null && levelExpression.NodeType == ExpressionType.MemberAccess && objectMapper.Associations.Count > 0 || inheritance.Count > 0)
+				if (tableAssociation == null && levelExpression.NodeType == ExpressionType.MemberAccess && (objectMapper.Associations.Count > 0 || inheritance.Count > 0))
```
Comment (MaceWindu) [inline Source/LinqToDB/Linq/Builder/TableBuilder.TableContext.cs]: Looks like github failed to build diff properly, I've changed only this line in this file

**Comment by MaceWindu** [comment]
@jogibear9988 , I've fixed incorrectly-typed `QueryExpressionMethod`'s (starting from RC0 it gives decsriptive error message) and now tests work (probably due to changes in associations in rc0)

@sdanyliv could you check association-based query - is it expected generation? It generates APLLY query (so it doesn't work with sqlite now)

**Pull request merged**